src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    static class Node<T> {
+        private T val;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T val, Node<T> prev) {
+            this.val = val;
+            this.prev = prev;
+        }
+
+        public Node(T value, Node<T> prev, Node<T> next) {
+            this.prev = prev;
+            this.val = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<T>(value, tail);
+        if (size == 0) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("The index " + index + " is invalid");
```

Comment:
you can extract method checkIndex() on this code `if (index > size || index < 0) {
             throw new IndexOutOfBoundsException("The index " + index + " is invalid");
         }`